# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/binutils/muslx32.patch
# Copyright (C) 2021 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- binutils-2.39/ld/configure.tgt.vanilla	2022-09-01 14:25:14.933770324 +0200
+++ binutils-2.39/ld/configure.tgt	2022-09-01 14:25:58.587767722 +0200
@@ -993,11 +993,11 @@
 x86_64-*-haiku*)	targ_emul=elf_x86_64_haiku
 			targ_extra_emuls="elf_x86_64 elf_i386_haiku"
 			;;
-x86_64-*-linux-gnux32)	targ_emul=elf32_x86_64
+x86_64-*-linux-*x32)	targ_emul=elf32_x86_64
 			targ_extra_emuls="elf_x86_64 elf_i386 elf_iamcu"
 			targ_extra_libpath="elf_i386 elf_iamcu elf_x86_64"
-			tdir_elf_iamcu=`echo ${targ_alias} | sed -e 's/x86_64/i386/' -e 's/-linux-gnux32/-linux-gnu/'`
-			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/' -e 's/-linux-gnux32/-linux-gnu/'`
+			tdir_elf_iamcu=`echo ${targ_alias} | sed -e 's/x86_64/i386/' -e 's/x32//'`
+			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/' -e 's/x32//'`
 			;;
 x86_64-*-linux-*)	targ_emul=elf_x86_64
 			targ_extra_emuls="elf32_x86_64 elf_i386 elf_iamcu"
--- binutils-2.37/gas/configure.tgt.vanilla	2021-10-18 14:49:18.207289232 +0200
+++ binutils-2.37/gas/configure.tgt	2021-10-18 14:49:49.502290847 +0200
@@ -231,7 +231,7 @@
   i386-*-openbsd*)			fmt=elf em=nbsd ;;
   i386-*-linux-*)			fmt=elf em=linux
     case ${cpu}-${os} in
-      x86_64*-linux-gnux32)		arch=x86_64:32 ;;
+      x86_64*-linux-*x32)		arch=x86_64:32 ;;
     esac ;;
   i386-*-lynxos*)			fmt=elf em=lynx ;;
   i386-*-redox*)			fmt=elf ;;
